### ISSUE TITLE
Support for `top_hits` Metrics Aggregation

### ISSUE THREAD

**Issue by bermanjosh** [bitemyapp/bloodhound#163]
The 'Top hits Aggregation' is a way of ordering results within a bucket. I believe I need it for a new use case I'm working on (grouping similar search results). Based on what I can tell, it differs from other aggregations in 2 ways:
1. It returns the document as part of the aggregation bucket
2. It isn't "nestable", that is, it's only supposed to be used as a lower-level aggregation, not a top-level, and it cannot have sub-aggregations. 

I'd like to add support for this, specifically, hitting the following criteria:
- It's `AggregationResults` instance returns the actual document / Haskell object. It seems to be similar to `Hit a`, so I'm hoping I can piggyback on that. 
- Nice to have - Enforce the lower-level requirement. 

I might add more here as I work through it, if that's ok. 

Edit - @bitemyapp, can you assign this to me?

**Comment by bermanjosh** [comment]
Tentative implamentation is in: https://github.com/bermanjosh/bloodhound/commits/badAggTest. Will PR after #158.
